File: insta/app/src/main/java/com/example/insta/News.java
package com.example.insta;


import android.os.Parcel;
import android.os.Parcelable;

public class News implements Parcelable {
    private String author;
    private String title;
    private String postData;
    private String photoLink;
    private String viewsCount;
    private String commentsCount;



    private String ava;
    private int postPhoto;
    private boolean postLiked = false;


    public News(
            String postData,
            String author,
            String title,
            String viewsCount,
            String commentsCount,
            String photoLink,
            String ava
    ) {
        this.photoLink = photoLink;
        this.postData = postData;
        this.author = author;
        this.title = title;
        this.viewsCount = viewsCount;
        this.commentsCount = commentsCount;
        this.ava = ava;


    }

    public boolean isPostLiked() {
        return postLiked;
    }

    public void setPostLiked(boolean postLiked) {
        this.postLiked = postLiked;
    }
    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }


    public String getViewsCount() {
        return viewsCount;
    }

    public void setViewsCount(String viewsCount) {
        this.viewsCount = viewsCount;
    }

    public String getCommentsCount() {
        return commentsCount;
    }

    public void setCommentsCount(String commentsCount) {
        this.commentsCount = commentsCount;
    }

    public String getAva() {
        return ava;
    }

    public void setAva(String ava) {
        this.ava = ava;
    }

    public int getPostPhoto() {
        return postPhoto;
    }

    public void setPostPhoto(int postPhoto) {
        this.postPhoto = postPhoto;
    }

    public String getPostData() {
        return postData;
    }

    public void setPostData(String postData) {
        this.postData = postData;
    }


    public String getPhotoLink() {
        return photoLink;
    }

    public void setPhotoLink(String photoLink) {
        this.photoLink = photoLink;
    }


    @Override
    public String toString() {
        return "News{" +
                "author='" + author + '\'' +
                ", title='" + title + '\'' +
                ", postData='" + postData + '\'' +
                ", viewsCount='" + viewsCount + '\'' +
                ", commentsCount='" + commentsCount + '\'' +
                ", ava='" + ava + '\'' +
                ", postPhoto=" + postPhoto +
                '}';
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.author);
        dest.writeString(this.title);
        dest.writeString(this.viewsCount);
        dest.writeString(this.commentsCount);

        dest.writeString(this.ava);
        dest.writeInt(this.postPhoto);

        dest.writeString(this.postData);
        dest.writeString(this.photoLink);

    }

    protected News(Parcel in) {
        this.author = in.readString();
        this.title = in.readString();
        this.viewsCount = in.readString();
        this.commentsCount = in.readString();

        this.ava = in.readString();
        this.postPhoto = in.readInt();
        this.postData = in.readString();

        this.photoLink = in.readString();
    }

    public static final Parcelable.Creator<News> CREATOR = new Parcelable.Creator<News>() {
        @Override
        public News createFromParcel(Parcel source) {

            return new News(source);
        }

        @Override
        public News[] newArray(int size) {

            return new News[size];
        }
    };
}

File: insta/app/src/main/java/com/example/insta/News.kt
package com.example.insta


import android.os.Parcel
import android.os.Parcelable

class News : Parcelable {
    var author: String? = null
    var title: String? = null
    var postData: String? = null
    var photoLink: String? = null
    var viewsCount: String? = null
    var commentsCount: String? = null


    var ava: String? = null
    var postPhoto: Int = 0
    var isPostLiked = false


    constructor(
            postData: String,
            author: String,
            viewsCount: String,
            title: String,
            commentsCount: String,
            photoLink: String,
            ava: String
    ) {
        this.photoLink = photoLink
        this.postData = postData
        this.author = author
        this.title = title
        this.viewsCount = viewsCount
        this.commentsCount = commentsCount
        this.ava = ava


    }


    override fun toString(): String {
        return "News{" +
                "author='" + author + '\''.toString() +
                ", title='" + title + '\''.toString() +
                ", postData='" + postData + '\''.toString() +
                ", viewsCount='" + viewsCount + '\''.toString() +
                ", commentsCount='" + commentsCount + '\''.toString() +
                ", ava='" + ava + '\''.toString() +
                ", postPhoto=" + postPhoto +
                '}'.toString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(this.author)
        dest.writeString(this.title)
        dest.writeString(this.viewsCount)
        dest.writeString(this.commentsCount)

        dest.writeString(this.ava)
        dest.writeInt(this.postPhoto)

        dest.writeString(this.postData)
        dest.writeString(this.photoLink)

    }

    private constructor(parcel: Parcel) {
        this.author = parcel.readString()
        this.title = parcel.readString()
        this.viewsCount = parcel.readString()
        this.commentsCount = parcel.readString()

        this.ava = parcel.readString()
        this.postPhoto = parcel.readInt()
        this.postData = parcel.readString()
        this.photoLink = parcel.readString()
    }

//    companion object {
//
//        val CREATOR: Parcelable.Creator<News> = object : Parcelable.Creator<News> {
//            override fun createFromParcel(source: Parcel): News {
//
//                return News(source)
//            }
//
//            override fun newArray(size: Int): Array<News> {
//
//                return arrayOfNulls(size)
//            }
//        }
//    }

    companion object CREATOR : Parcelable.Creator<News> {
            override fun createFromParcel(parcel: Parcel): News {
                return News(parcel)
            }

        override fun newArray(size: Int): Array<News?> {
            return arrayOfNulls(size)
        }
    }
}